### PR TITLE
Upgrade to as-pect 2.7.0 in anticipation of AssemblyScript 0.9 release

### PR DESCRIPTION
**Pull request by jtenner**
Please help me find more optimizations and I would be glad to apply them anywhere function pointers need to be compared.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,18 +8,17 @@
 		"build:js": "mkdir -p build/ && cp -R src/js build/",
 		"build:html": "mkdir -p build/ && cp -R src/index.html build/",
 		"build:infra": "mkdir -p build/ && tsc -p src/infra/tsconfig.json",
-		"build": "rm -rf build && npm run build:infra && npm run build:html && npm run build:js && npm run build:ts && npm run build:as",
+		"build": "rimraf build && npm run build:infra && npm run build:html && npm run build:js && npm run build:ts && npm run build:as",
```
Comment (trusktr) [inline package.json]: Is the point of this to make it cross-platform (i.e. to work in Windows)?

Comment (jtenner) [reply package.json]: Yeah I was thinking about that, but I can remove it if it doesn't make sense. This is an artifact from my local setup and will clean it up later.

File: src/as/__tests__/as-pect.d.ts [hunk 2]
```diff
@@ -1,482 +1 @@
...
- */
-declare function reportMin(value: bool): void
-
-/**
- * This function call enables gathering the varaince of the samples collected for the following test.
- *
- * @param {bool} enabled - The bool to indicate if the variance should be calculated.
- */
-declare function reportVariance(value: bool): void
+/// <reference path="../../../node_modules/@as-pect/assembly/types/as-pect.d.ts" />
```
Comment (trusktr) [inline src/as/__tests__/as-pect.d.ts]: I haven't used `reference` comments before. Is this better than adding the file in tsconfig `files`?

Comment (jtenner) [reply src/as/__tests__/as-pect.d.ts]: All I did was run `npx asp --types` to clean up the types declaration. There's no reason to expose those types now that `as-pect` is more mature.

This was the easiest solution I could come up with to ensure good ide typings.

As for placing it in the `files` for tsconfig, if you prefer that, I can make a commit to help with that.

File: package.json (hunk 1 above)
Comment (trusktr) [reply package.json]: All good. Eventually I'd like for it to work in Windows too.

Comment (jtenner) [reply package.json]: I believe I had a problem with the `mkdir -p` commands before I gave up trying to get everything to work. Not sure what to do about that :)

**Comment by jtenner** [comment]
Can we do a seperate pull request to help make things work cross platform?

**Comment by trusktr** [comment]
> Please help me find more optimizations and I would be glad to apply them anywhere function pointers need to be compared.

Off the top of my head I don't remember any places to update, and am planning to move onto porting new files and leaving whatever works as-is. But I'll use this on the new stuff now that I know I can. Thanks!

**Comment by trusktr** [comment]
> Can we do a seperate pull request to help make things work cross platform?

Yeah, totally. I'm not planning to do anything about that until I already have something rendering. I don't mind keeping rimraf here though.

**Comment by trusktr** [comment]
Are you able to hit merge on your end?

File: src/as/__tests__/as-pect.d.ts (hunk 2 above)
Comment (trusktr) [reply src/as/__tests__/as-pect.d.ts]: All good. Was just curious. it works. :)

**Pull request merged**